src/dvc_objects/db.py

```diff
@@ -216,7 +216,8 @@ def _list_prefixes(
         yield from self.fs.find(paths, batch_size=jobs, prefix=prefix)
 
     def path_to_oid(self, path) -> str:
-        parts = self.fs.path.parts(path)[-2:]
+        self_parts = self.fs.path.parts(self.path)
+        parts = self.fs.path.parts(path)[len(self_parts) :]
```

Comment:
This may eventually need to be a hard check to make sure that `path` actually overlaps with the ODB path but I left it out for now